iceoryx_posh/include/iceoryx_posh/internal/runtime/runnable_data.hpp

```diff
@@ -23,8 +23,9 @@ namespace iox
 namespace runtime
 {
 /// @brief struct which contains all the members of an object of type Runnable
-struct RunnableData
+class RunnableData
```

Comment:
There was a mixture of `class RunnableData` and `struct RunnableData` in multiple forward declarations. I think the confusion and mixture would be solved if this becomes a class since most developers are assuming that this is a class when they declare it.